net/ceph/osd_client.c

```diff
@@ -771,11 +837,13 @@ static u32 osd_req_encode_op(struct ceph_osd_op *dst,
 		break;
 	case CEPH_OSD_OP_STARTSYNC:
 		break;
-	case CEPH_OSD_OP_NOTIFY_ACK:
 	case CEPH_OSD_OP_WATCH:
 		dst->watch.cookie = cpu_to_le64(src->watch.cookie);
-		dst->watch.ver = cpu_to_le64(src->watch.ver);
-		dst->watch.flag = src->watch.flag;
+		dst->watch.ver = cpu_to_le64(0);
+		dst->watch.op = src->watch.op;
+		dst->watch.gen = cpu_to_le32(src->watch.gen);
+		break;
+	case CEPH_OSD_OP_NOTIFY_ACK:
```

Comment:
Hmm, I had to add it. Maybe one of my cherry-picks took it out?

> On May 25, 2016, at 2:12 PM, Ilya Dryomov <notifications@github.com> wrote:
> 
> In net/ceph/osd_client.c:
> 
> >  	case CEPH_OSD_OP_WATCH:
> >  		dst->watch.cookie = cpu_to_le64(src->watch.cookie);
> > -		dst->watch.ver = cpu_to_le64(src->watch.ver);
> > -		dst->watch.flag = src->watch.flag;
> > +		dst->watch.ver = cpu_to_le64(0);
> > +		dst->watch.op = src->watch.op;
> > +		dst->watch.gen = cpu_to_le32(src->watch.gen);
> > +		break;
> > +	case CEPH_OSD_OP_NOTIFY_ACK:
> 
> case CEPH_OSD_OP_NOTIFY is added in "libceph: support for sending notifies". Apparently github can't keep the order of commits straght or something...
> 
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
>